test/helpers/foundry/types.ts

```diff
@@ -0,0 +1,100 @@
+import type { Agent as HttpsAgent } from 'node:https';
+import type { Agent as HttpAgent } from 'node:http';
+import type { InferredOptionTypes, Options } from 'yargs';
+
+// #region utils
+
+type UnionToIntersection<U> = ((k: U) => void) extends (k: infer I) => void
+  ? I
+  : never;
+
+type LastInUnion<U extends PropertyKey> = UnionToIntersection<
+  U extends PropertyKey ? () => U : never
+> extends () => infer Last
+  ? Last
+  : never;
+
+type UnionToTuple<U extends PropertyKey, Last = LastInUnion<U>> = [U] extends [
+  never,
+]
+  ? []
+  : [...UnionToTuple<Exclude<U, Last>>, Last];
+
+// #endregion utils
+
+// #region enums
+
+export enum Architecture {
+  Amd64 = 'amd64',
+  Arm64 = 'arm64',
+}
+
+export enum Extension {
+  Zip = 'zip',
+  Tar = 'tar.gz',
+}
+
+export enum Platform {
+  Windows = 'win32',
+  Linux = 'linux',
+  Mac = 'darwin',
+}
+
+export enum Binary {
+  Anvil = 'anvil',
+  Forge = 'forge',
+  Cast = 'cast',
+  Chisel = 'chisel',
```

Comment:
we could use all these just by adding to `--binaries <binary you want>` to the `yarn foundryup` command. neat!